js/actions/actions.js

```diff
@@ -21,27 +22,33 @@ module.exports = {
     var message = {
       text: text,
       timestamp: timestamp,
-      user: 'guest' // hard-code for now
+      user: 'guest', // hard-code for now
+      clientId: UUID.generate()
     };
 
+    dispatch(ActionTypes.NEW_MESSAGE, message);
+
     QuackData.create('message', {
       message: message,
       success: function() {
-        dispatch(ActionTypes.CREATE_MESSAGE_SUCCESS);
+        dispatch(ActionTypes.CREATE_MESSAGE_SUCCESS, message);
       },
       error: function(err) {
-        dispatch(ActionTypes.CREATE_MESSAGE_ERROR);
+        dispatch(
+          ActionTypes.CREATE_MESSAGE_ERROR,
+          {message: message, error: err}
```

Comment:
With your new dispatch helper the two lines rule sort of looks weird to me. Changed though.